-1 
Tested at: 5f95bc2029a2b2c96df08d3e3ff6904d571abe8c
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 8 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-07-08-2300/src/RecoTracker/TkHitPairs/src/InnerDeltaPhi.cc:22:17: warning: unused function 'checked_asin' [-Wunused-function]
  inline double checked_asin(double x, const char *expr, const char *file, int line) {
                ^
1 warning generated.
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-07-08-2300/src/RecoTracker/TrackProducer/src/TrackProducerAlgorithm.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-07-08-2300/src/RecoTracker/TkHitPairs/src/HitPairGeneratorFromLayerPair.cc:139:15: error: variable-sized object may not be initialized
      bool ok[e-b]{};
              ^~~
1 error generated.
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-07-08-2300/src/RecoTracker/TrackProducer/src/TrajectoryToResiduals.cc 
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-07-08-2300/src/RecoTracker/TkTrackingRegions/src/SealModule.cc:5:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10066/6084/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b249178a4b1948d1d408753cae2f9fb1f8e91238
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10066/6084/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10066/6084/git-merge-result

--- a/DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc
+++ b/DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc
@@ -610,7 +610,7 @@ void SiStripMonitorTrack::trajectoryStudy(const edm::Ref<std::vector<Trajectory>
       LogTrace("SiStripMonitorTrack")<<"\nMatched recHit found"<< std::endl;
       //	type=Matched;
 
-      GluedGeomDet * gdet=(GluedGeomDet *)tkgeom_->idToDet(matchedhit->geographicalId());
+      const GluedGeomDet * gdet=static_cast<const GluedGeomDet *>(tkgeom_->idToDet(matchedhit->geographicalId()));
       GlobalVector gtrkdirup=gdet->toGlobal(updatedtsos.localMomentum());
       //mono side
       const GeomDetUnit * monodet=gdet->monoDet();
@@ -628,7 +628,7 @@ void SiStripMonitorTrack::trajectoryStudy(const edm::Ref<std::vector<Trajectory>
     else if(projhit){
       LogTrace("SiStripMonitorTrack")<<"\nProjected recHit found"<< std::endl;
       //	type=Projected;
-      GluedGeomDet * gdet=(GluedGeomDet *)tkgeom_->idToDet(projhit->geographicalId());
+      const GluedGeomDet * gdet=static_cast<const GluedGeomDet *>(tkgeom_->idToDet(projhit->geographicalId()));
 
       GlobalVector gtrkdirup=gdet->toGlobal(updatedtsos.localMomentum());
       const SiStripRecHit2D  originalhit=projhit->originalHit();
@@ -674,7 +674,7 @@ void SiStripMonitorTrack::hitStudy(const edm::EventSetup& es,
   if(matchedhit){     // type=Matched;
     LogTrace("SiStripMonitorTrack")<<"\nMatched recHit found"<< std::endl;    
 
-    GluedGeomDet * gdet=(GluedGeomDet *)tkgeom_->idToDet(matchedhit->geographicalId());
+    const GluedGeomDet * gdet=static_cast<const GluedGeomDet *>(tkgeom_->idToDet(matchedhit->geographicalId()));
 
     GlobalVector gtrkdirup=gdet->toGlobal(localMomentum);
 
@@ -693,7 +693,7 @@ void SiStripMonitorTrack::hitStudy(const edm::EventSetup& es,
   else if(projhit){    // type=Projected;
       LogTrace("SiStripMonitorTrack")<<"\nProjected recHit found"<< std::endl;      
 
-      GluedGeomDet * gdet=(GluedGeomDet *)tkgeom_->idToDet(projhit->geographicalId());      
+      const GluedGeomDet * gdet=static_cast<const GluedGeomDet *>(tkgeom_->idToDet(projhit->geographicalId()));
 
       GlobalVector gtrkdirup=gdet->toGlobal(localMomentum);
       const SiStripRecHit2D  originalhit=projhit->originalHit();
@@ -1034,7 +1034,7 @@ void SiStripMonitorTrack::fillModMEs(SiStripClusterInfo* cluster,std::string nam
     // https://indico.cern.ch/event/342236/session/5/contribution/10/material/slides/0.pdf
     float dQdx_fromTrack = siStripClusterTools::chargePerCM(detid, *cluster, LV);
     // from straigth line origin-sensor centre
-    const StripGeomDetUnit* DetUnit = (const StripGeomDetUnit*) tkgeom_->idToDetUnit(DetId(detid));
+    const StripGeomDetUnit* DetUnit = static_cast<const StripGeomDetUnit*>(tkgeom_->idToDetUnit(DetId(detid)));
     LocalPoint locVtx = DetUnit->toLocal(GlobalPoint(0.0, 0.0, 0.0));
     LocalVector locDir(locVtx.x(), locVtx.y(), locVtx.z());
     float dQdx_fromOrigin = siStripClusterTools::chargePerCM(detid, *cluster, locDir);

--- a/RecoTracker/FinalTrackSelectors/plugins/TrackerTrackHitFilter.cc
+++ b/RecoTracker/FinalTrackSelectors/plugins/TrackerTrackHitFilter.cc
@@ -420,7 +420,6 @@ TrackerTrackHitFilter::produce(edm::Event &iEvent, const edm::EventSetup &iSetup
 	if(nvalidhits >= int(minimumHits_)){
 	  output->push_back( makeCandidate ( *trk, begin, end ) );
 	}
-	nvalidhits=0;
       }
       else{//all invalid hits have been already kicked out
 	if ((end - begin) >= int(minimumHits_)) {

--- a/RecoTracker/SingleTrackPattern/src/CRackTrajectoryBuilder.cc
+++ b/RecoTracker/SingleTrackPattern/src/CRackTrajectoryBuilder.cc
@@ -665,7 +665,7 @@ void CRackTrajectoryBuilder::AddHit(Trajectory &traj,
                double currChi2 = theEstimator->estimate(prSt, *tmpHit).second;
                if ( currChi2 < chi2min )
                  {
-                   currChi2 = chi2min;
+                   chi2min = currChi2;
                    bestHit = tmpHit;
                  }
              }

--- a/RecoVertex/BeamSpotProducer/src/BSFitter.cc
+++ b/RecoVertex/BeamSpotProducer/src/BSFitter.cc
@@ -742,7 +742,6 @@ for( iparam = fBSvector.begin(); iparam != fBSvector.end(); ++iparam)
                                            tracksfixed++; }
 
                        function = function + log(tot_pdf);
-                       tot_pdf=0.0;
 
 
        }//loop over tracks

--- a/SimGeneral/TrackingAnalysis/plugins/TrackingTruthAccumulator.cc
+++ b/SimGeneral/TrackingAnalysis/plugins/TrackingTruthAccumulator.cc
@@ -1105,15 +1105,14 @@ namespace // Unnamed namespace for things only used in this file
 		if( pTrackingParticle==NULL )
 		{
 			// Need to make sure the production vertex has been created first
-			TrackingVertex* pProductionVertex=pOutput->getTrackingVertex( pDecayTrack->pParentVertex );
-			if( pProductionVertex==NULL )
+			if( pOutput->getTrackingVertex( pDecayTrack->pParentVertex ) == nullptr )
 			{
 				// TrackingVertex doesn't exist in the output collection yet. However, it's already been
 				// created in the addTrack() function and a temporary reference to it set in the TrackingParticle.
 				// I'll use that reference to create a copy in the output collection. When the TrackingParticle
 				// is added to the output collection a few lines below the temporary reference to the parent
 				// vertex will be cleared, and the correct one referring to the output collection will be set.
-				pProductionVertex=pOutput->addTrackingVertex( pDecayTrack->pParentVertex, *trackingParticle.parentVertex() );
+				pOutput->addTrackingVertex( pDecayTrack->pParentVertex, *trackingParticle.parentVertex() );
 			}
 
 

--- a/SimTracker/TrackerHitAssociation/src/TrackerHitAssociator.cc
+++ b/SimTracker/TrackerHitAssociation/src/TrackerHitAssociator.cc
@@ -99,21 +99,12 @@ void TrackerHitAssociator::makeMaps(const edm::Event& theEvent, const TrackerHit
   //  The collections are specified via ROUList in the configuration, and can
   //  be either crossing frames (e.g., mix/g4SimHitsTrackerHitsTIBLowTof)
   //  or just PSimHits (e.g., g4SimHits/TrackerHitsTIBLowTof)
-  const char* const highTag = "HighTof";
-  unsigned int tofBin; 
-  edm::EDConsumerBase::Labels labels;
   if (assocHitbySimTrack_) {
     for(auto const& cfToken : config.cfTokens_) {
       edm::Handle<CrossingFrame<PSimHit> > cf_simhit;
       //int Nhits = 0;
       if (theEvent.getByToken(cfToken, cf_simhit)) {
         std::unique_ptr<MixCollection<PSimHit> > thisContainerHits(new MixCollection<PSimHit>(cf_simhit.product())); 
-        theEvent.labelsForToken(cfToken, labels);
-        if(std::strstr(labels.productInstance, highTag) != NULL) {
-          tofBin = StripDigiSimLink::HighTof;
-        } else {
-          tofBin = StripDigiSimLink::LowTof; 
-        }    
         for (auto const& isim : *thisContainerHits) {
           DetId theDet(isim.detUnitId());
           SimHitMap[theDet].push_back(isim);
@@ -126,12 +117,6 @@ void TrackerHitAssociator::makeMaps(const edm::Event& theEvent, const TrackerHit
       edm::Handle<std::vector<PSimHit> > simHits;
       //int Nhits = 0;
       if(theEvent.getByToken(simHitToken, simHits)) {
-        theEvent.labelsForToken(simHitToken, labels);
-        if(std::strstr(labels.productInstance, highTag) != NULL) {
-          tofBin = StripDigiSimLink::HighTof;
-        } else {
-          tofBin = StripDigiSimLink::LowTof; 
-        }
         for (auto const& isim : *simHits) {
           DetId theDet(isim.detUnitId());
           SimHitMap[theDet].push_back(isim);
@@ -141,6 +126,9 @@ void TrackerHitAssociator::makeMaps(const edm::Event& theEvent, const TrackerHit
       }
     }
   } else {
+    const char* const highTag = "HighTof";
+    unsigned int tofBin;
+    edm::EDConsumerBase::Labels labels;
     simHitCollectionID theSimHitCollID;
     for(auto const& cfToken : config.cfTokens_) {
       edm::Handle<CrossingFrame<PSimHit> > cf_simhit;

--- a/TrackPropagation/SteppingHelixPropagator/src/SteppingHelixPropagator.cc
+++ b/TrackPropagation/SteppingHelixPropagator/src/SteppingHelixPropagator.cc
@@ -914,25 +914,21 @@ bool SteppingHelixPropagator::makeAtomStep(SteppingHelixPropagator::StateInfo& s
 
     double oneLessCosPhi=0;
     double oneLessCosPhiOPhi=0;
-    double sinPhiOPhi=0;
     double phiLessSinPhiOPhi=0;
 
     if (phiSmall){
       double phi2 = phi*phi;
       double phi3 = phi2*phi;
       double phi4 = phi3*phi;
-      sinPhi = phi - phi3/6. + phi4*phi/120.;
-      cosPhi = 1. -phi2/2. + phi4/24.;
       oneLessCosPhi = phi2/2. - phi4/24. + phi2*phi4/720.; // 0.5*phi*phi;//*(1.- phi*phi/12.);
       oneLessCosPhiOPhi = 0.5*phi - phi3/24. + phi2*phi3/720.;//*(1.- phi*phi/12.);
-      sinPhiOPhi = 1. - phi*phi/6. + phi4/120.;
       phiLessSinPhiOPhi = phi*phi/6. - phi4/120. + phi4*phi2/5040.;//*(1. - phi*phi/20.);
     } else {
       cosPhi = cos(phi);
       sinPhi = sin(phi);
       oneLessCosPhi = 1.-cosPhi;
       oneLessCosPhiOPhi = oneLessCosPhi/phi;
-      sinPhiOPhi = sinPhi/phi;
+      double sinPhiOPhi = sinPhi/phi;
       phiLessSinPhiOPhi = 1 - sinPhiOPhi;
     }
 
@@ -1009,14 +1005,13 @@ bool SteppingHelixPropagator::makeAtomStep(SteppingHelixPropagator::StateInfo& s
       cosPhi = 1. -phi2/2. + phi4/24.;
       oneLessCosPhi = phi2/2. - phi4/24. + phi2*phi4/720.; // 0.5*phi*phi;//*(1.- phi*phi/12.);
       oneLessCosPhiOPhi = 0.5*phi - phi3/24. + phi2*phi3/720.;//*(1.- phi*phi/12.);
-      sinPhiOPhi = 1. - phi*phi/6. + phi4/120.;
       phiLessSinPhiOPhi = phi*phi/6. - phi4/120. + phi4*phi2/5040.;//*(1. - phi*phi/20.);
     }else {
       cosPhi = cos(phi); 
       sinPhi = sin(phi);
       oneLessCosPhi = 1.-cosPhi;
       oneLessCosPhiOPhi = oneLessCosPhi/phi;
-      sinPhiOPhi = sinPhi/phi;
+      double sinPhiOPhi = sinPhi/phi;
       phiLessSinPhiOPhi = 1. - sinPhiOPhi;
     }
 

--- a/TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc
+++ b/TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc
@@ -203,18 +203,13 @@ TrackDetMatchInfo TrackDetectorAssociator::associate( const edm::Event& iEvent,
    cachedTrajectory_.setMinDetectorRadius(minR);
    cachedTrajectory_.setMinDetectorLength(minZ*2.);
 
-   double maxR(0);
-   double maxZ(0);
-
    if (parameters.useMuon) {
-     maxR = muonDetIdAssociator_->volume().maxR();
-     maxZ = muonDetIdAssociator_->volume().maxZ();
+     double maxR = muonDetIdAssociator_->volume().maxR();
+     double maxZ = muonDetIdAssociator_->volume().maxZ();
      cachedTrajectory_.setMaxDetectorRadius(maxR);
      cachedTrajectory_.setMaxDetectorLength(maxZ*2.);
    }
    else {
-     maxR = HOmaxR;
-     maxZ = HOmaxZ;
      cachedTrajectory_.setMaxDetectorRadius(HOmaxR);
      cachedTrajectory_.setMaxDetectorLength(HOmaxZ*2.);
    }

--- a/Validation/RecoVertex/src/V0Validator.cc
+++ b/Validation/RecoVertex/src/V0Validator.cc
@@ -438,10 +438,8 @@ void V0Validator::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
     }
   }
   nKs->Fill( (float) numK0sFound );
-  numK0sFound = 0.;
 
   double numLamFound = 0.;
-  mass = 0.;
   radDist.clear();
   // Lambdas
   if ( lambdaCollection->size() > 0 ) {
@@ -562,7 +560,6 @@ void V0Validator::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
     }
   }
   nLam->Fill( (double) numLamFound );
-  numLamFound = 0.;
 
 
   ///////////////////////////////

